src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,109 @@
-import { PeopleFilters } from './PeopleFilters';
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Person } from '../types';
+import { getPeople } from '../api';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { PeopleFilters } from './PeopleFilters';
+import { useSearchParams } from 'react-router-dom';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+  const { slug } = useParams<{ slug: string }>();
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(peopleFromServer => {
+        setPeople(peopleFromServer);
+      })
+      .catch(() => {
+        setError('Something went wrong');
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const [searchParams] = useSearchParams();
+  const sortField = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const currentSex = searchParams.get('sex');
+  const queryParam = searchParams.get('query') || '';
+  const selectedCenturies = searchParams.getAll('centuries');
+
+  const sortedPeople = [...people];
+
+  if (sortField) {
+    sortedPeople.sort((a, b) => {
+      const aVal = a[sortField as keyof Person];
+      const bVal = b[sortField as keyof Person];
+
+      if (aVal == null || bVal == null) {
+        return 0;
+      }
+
+      if (aVal > bVal) {
+        return order === 'desc' ? -1 : 1;
+      }
+
+      if (aVal < bVal) {
+        return order === 'desc' ? 1 : -1;
+      }
+
+      return 0;
+    });
+  }
+
+  const filteredPeople = sortedPeople.filter(person => {
+    const matchesQuery = queryParam
+      ? person.name.toLowerCase().includes(queryParam.toLowerCase())
```

Comment:
In this case you also need to show people who have mother or father with the same query